### ISSUE TITLE
Dates are dangerously built

### ISSUE THREAD

**Issue by bejoinka** [awell-health/awell-extensions#98]
There's a local/UTC mismatch somewhere in this `stringDate` function (assuming it's in the parseISO function, but unsure)

```ts
import { stringDate } from '../../validation/generic.zod'
import { z } from 'zod'

describe('formatISO', () => {
  test('test formatISO date', async () => {
    const date = '2023-01-01'
    const zodDate = z.object({ date: stringDate })
    const parsed = zodDate.parse({ date })
    expect(parsed).toStrictEqual({ date })
  })
})

```
returns
```bash
$ yarn test -o
 FAIL  extensions/healthie/actions/__tests__/zod.ts
  formatISO
    ✕ test formatISO date (5 ms)

  ● formatISO › test formatISO date

    expect(received).toStrictEqual(expected) // deep equality

    - Expected  - 1
    + Received  + 1

      Object {
    -   "date": "2023-01-01",
    +   "date": "2022-12-31",
      }

       7 |     const zodDate = z.object({ date: stringDate })
       8 |     const parsed = zodDate.parse({ date })
    >  9 |     expect(parsed).toStrictEqual({ date })
         |                    ^
      10 |   })
      11 | })
      12 |

      at Object.<anonymous> (extensions/healthie/actions/__tests__/zod.ts:9:20)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        3.7 s, estimated 4 s
```

**Comment by bejoinka** [comment]
this issue appears to be resolved. i'm confused, though, because the function in `DateSchema` is the same as the `stringDate` function that was in `generic.zod.ts`.

I'm going to close this ticket for now. Would appreciate five minutes of review from @michal-grzelak just to confirm. Thank you!

**Comment by michal-grzelak** [comment]
@bejoinka 

TLDR: It's just a local issue I fixed by forcing `jest` to always run in the `UTC` timezone.
---

I can explain because I stumbled upon it too. It is the cause when you test it locally. When you test locally, jest by default takes the timezone of the local computer. When deployed, all servers are always set up to use UTC/GMT timezone, hence there is no problem.
See jest.config.js, in the last PR I added
```
// Make sure jest runs with UTC TZ (to avoid failed tests locally)
process.env.TZ = 'UTC'
```
So that jest is always run with UTC timezone. Without it, every person from different country in that project would get different test result.
